### ISSUE TITLE
Running composer update not pulling in middleware changes 

### ISSUE THREAD

**Issue by dakala** [Zizaco/entrust#391]
Hi. I've been trying to update this package as the middleware changes got merged a couple of days ago. For the life of me, it doesn't update even though composer seems happy each time I ran it.

"require": {

		"zizaco/entrust": "dev-laravel-5",
        "laravelcollective/html": "5.1.*",
        "nesbot/carbon": "~1.14",
        "cviebrock/eloquent-sluggable": "dev-master",
        "fzaninotto/faker": "~1.4",
        "dakala/laravel5-setting": "dev-master",
        "league/csv": "^7.1@dev"
    },

How do I update zizaco entrust to pull in the latest changes, please?

**Comment by dakala** [comment]
I did a full composer reset - backed up my vendor/ and composer.lock and ran composer install. That didn't help. 

Then I changed the version to "dev-master" as the laravel-5 branch doesn't have the middleware changes. After another full reset, middleware got pulled but no tests/ folder. To make composer happy, I cloned the master branch somewhere and copied the tests folder across. After another composer update, everything seems to be happy.

I don't know if something's broken at my end but composer update should have pulled the entire package for me.

**Comment by lesterchan** [comment]
I am having the same exact issue!

**Comment by dakala** [comment]
So it isn't just me then :-) @lesterchan: Have you tried my fix? Give it a go please.

**Comment by lesterchan** [comment]
I know your solution work, but I will wait for the official fix. If it takes too long, I will fallback to laravel-5 and copy out the Middleware to my own since I need just the Role Middleware.

**Comment by dakala** [comment]
Yeah, that's an option I'm even considering now. Prior to this merge, I'd already applied the original patch with some modification as I didn't want to do abort(403) or something. 

If in the next couple of days master or laravel-5 branch isn't fixed, I'll probably revert myself.

**Comment by kieranja** [comment]
Caused me a bit of a headache - Could do a PR if it'd help?

**Comment by bahjaat** [comment]
Gimme a call when there is an update :)

**Comment by oldskool73** [comment]
:+1: 

As per my comment on the commit above, this is because of https://github.com/Zizaco/entrust/pull/389 , which adds the test file to the composer autoloads (seems like a bad move to me...?), but the zip doesn't include the tests dir, so composer fails.

A temporary workaround is to force the source to be used instead of the zip:

```
rm -rf vendor/zizaco/entrust
composer install --prefer-source
```
 Hopefully someone will revert the commit and fix this, as currently master is completely broken.

**Comment by tuwannu** [comment]
Interesting, I use "zizaco/entrust": "dev-master" in my project's composer.json and seems to work fine. Composer downloaded tests folder for me so it works on my machine.

Anyhow, I'm making a PR for this fix in a minute.

**Comment by tuwannu** [comment]
Meanwhile waiting for PR to be accepted. Here's a workaround by adding "repositories" and changing zizaco/entrust to use `dev-middlewaretest-fix`.

    {
        ...
        "repositories": [
            {
                "url": "https://github.com/tuwannu/entrust.git",
                "type": "git"
            }
        ],
        "require": {
            ...
            "zizaco/entrust": "dev-middlewaretest-fix",
            ...
        },
        ...
    }

**Comment by tuwannu** [comment]
This problem should already be resolved since #397.

Request closing this issue.
